sim/pkg/model/autoscaler_ticktock.go

```diff
@@ -98,13 +109,55 @@ func (asts *autoscalerTicktockStock) Add(entity simulator.Entity) error {
 	} else {
 		// do nothing
 	}
+}
 
-	//calculate CPU utilization
-	asts.calculateCPUUtilization()
+func (asts *autoscalerTicktockStock) adjustVertically(currentTime *time.Time) {
+	recommendedPodResources, err := asts.env.Plugin().VerticalRecommendation(currentTime.UnixNano())
+	if err != nil {
+		panic(err)
+	}
+	//create a map podName -> []recommendedPodResources
+	podToRecommendations := make(map[string][]*proto.RecommendedPodResources, 0)
+	for _, recommendation := range recommendedPodResources {
+		podToRecommendations[recommendation.PodName] = append(podToRecommendations[recommendation.PodName], recommendation)
+	}
 
-	return nil
+	//Iterate through replicas
+	pods := asts.cluster.ActiveStock().EntitiesInStock()
+	for _, pod := range pods {
+		//We have recommendations for the replica
+		for _, recommendation := range podToRecommendations[string((*pod).Name())] {
```

Comment:
You should do away with this for loop.  Just apply the same recommendations to all pods.